subgraph/src/fleek-nfa.ts

```diff
@@ -153,6 +156,89 @@ export function handleCollectionRoleRevoked(
   }
 }
 
+export function handleNewMint(event: NewMintEvent): void {
+  let newMintEntity = new NewMint(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  );
+
+  let name = event.params.name;
+  let description = event.params.description;
+  let externalURL = event.params.externalURL;
+  let ENS = event.params.ENS;
+  let gitRepository = event.params.gitRepository;
+  let commitHash = event.params.commitHash;
+  let logo = event.params.logo;
+  let color = event.params.color;
+  let accessPointAutoApprovalSettings =
+    event.params.accessPointAutoApprovalSettings;
+  let tokenId = event.params.tokenId;
+  let ownerAddress = event.params.owner;
+
+  newMintEntity.tokenId = tokenId;
```

Comment:
As per the comment above, we can remove the mint entity and just go directly from mint event to updating the token entity right?

Reply:
Yes, but why wouldn't we want to keep the record of the first mint? The best practice is to keep records of all events from the contract in subgraph.